Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionSidebarCocoa.mm

```diff
@@ -101,8 +102,9 @@
     if (!m_context.ptr() || isDefaultSidebar())
         return std::nullopt;
 
-    return m_tab.and_then([this](auto const& tab) { return m_context->getSidebar(*tab->window()); })
-        .value_or(m_context->defaultSidebar());
+    return m_tab.and_then([this](auto const& tab) -> std::optional<Ref<WebExtensionSidebar>> {
+            return tab->window() ? m_context->getSidebar(*(tab->window())) : std::nullopt;
```

Comment:
i think `m_extensionContext` would be a more precise name

Reply:
or rather or `extensionContext()` method

Reply:
and also, why does that method return an optional context?

Reply:
It returns an `optional<WebExtensionContext>` since we hold a weak pointer to the current context, so we need to check if it's still there on every access -- I just use the optional to indicate whether it's still there. In this particular instance, I'm using the member variable to access context rather than the getter because I check to ensure that context is still there at the beginning of this method, so there's no need to re-run the check by accessing context through the getter later on.